workflow/parser.py

```diff
@@ -0,0 +1,67 @@
+import os
+import subprocess
+
+
+class Parser:
+    def __init__(self, params):
+        self.params = params
+
+    def parse(self):
+
+        '''
+        - parse input config file 
+        - build TPS call 
+        - call TPS 
+        - gather outputs 
+        - return:
+            - list of outputs 
+            - output_folder
+            - out_label
+
+      
+        '''
+        # # grab TPS self.params 
+        build = []
+        OUT_FOLDER = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        OUT_LABEL = ""
+
+        # add TPS run call as first arg
+        build.extend(["bash", "./scripts/run"])
+
+        # grab self.params from config
+        req = self.params[1]["TPS"][0]["required"]
+        op = self.params[1]["TPS"][1]["optional"]
+
+        # filter given args
+        filtered = {**req, **op}
```

Comment:
Do we want to handle required and optional arguments separately?  Right now it seems like they are being split to let the user know what values they need to provide, right?

Reply:
Here I grab all values that are provided, then I loop through the provided values and deal with keys with no value (Boolean), prefix labels and output directories.